hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,202 @@ This document aims to make the distinction between physical entropy and factory
 However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
 That is either physical entropy, or the output of a DRNG which itself has been seeded (and possibly reseeded) with physical entropy.
 In CC terms, "entropy strings" (when used in this document without a qualifier) should be understood to come from either a PTG.2 or PTG.3 class RNG.
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+
+## Block Diagram
+
+
+
+## Hardware Interfaces
+
+
+
+## Design Details
+
+### Application Interface
+
+This section describes the application interface.
+The general interface operates like a FIFO, in which a command is pushed into the interface.
+An optional stream of data may follow, such as seed material for an instantiate command.
+There is one instance of a software interface.
+Many instances of the hardware interface maybe exist, and is determined by the NHwApp parameter.
+
+#### General Command Format
+
+Below is the general format for the application interface commands.
+The following fields are expanded below for clarity.
+
+- `acmd` : application command - selects one of five operations to perform
+- `shid` : state handle identification - selects which csrng maintained state to use
+- `xcnt` : transfer count - sets the total length of the transfer in beats, where zero is 1 beat
+- `flgs` : flags - used to allow for additional features
+- `creq` : count request - used only for the `generate` command, sets the number of genbits bits to be recieved
+
+
+
+                                                
+#### Command Header                                
+The application command header is 32 bits wide. Below is a description of the fields of this header:
+
+```
+  31          23          15            7             0
+  +-----------+-----------+------+------+------+------+
+  +          creq         | flgs | xcnt | shid | acmd |
+  +-----------+-----------+------+------+------+------+
+
+
+```
+
+
+- Total header: bits 31:0
+    - acmd: bits 2-0 
+        - 0x0: instantiate command
+        - 0x1: reseed command  
+        - 0x2: generate command
+        - 0x3: update command
+        - 0x4: uinstantiate command
+    - shid: bits 7-4
+        - supports up to 16 state handle ids
+    - xcnt: bits 11-8
+        - Number of 32b words to transfer
+    - flgs: bits 15-11
+        - flags for features
+    - creq: bits 31-16
+        - Number 128b genbits beats requested
+
+
+#### Specific Command Examples
+
+- instantiate:
+    - acmd: 0x0 - encoded command
```

Comment:
for all of these examples, I think I might suggest using Verilog syntax instead of C?  (4'h0, 4'h3, 4'hc, etc. for this example)

It's especially important if there are any non-nibble aligned subfields.

Reply:
reasonable - all 0x0 notation replaced with verilog notation.